### DISCUSSION TITLE
有关从TP同步到AP表的问题 (Questions about synchronizing from TP to AP table)

### DISCUSSION THREAD

**Discussion by Mr-Foyego** [stoneatom/stonedb#160] [Q&A] [answered]
请问：假设我在TP中创建了一个表testA，并指明engine=innodb，在进行一些业务写入操作后，这张表是否会同步到AP中？（相当于在TP中的表在AP中也有一份）
如果我要对testA进行分析，是不是需要等TP同步之后才能进行分析？
在TP中的表是不是全部会同步到AP中？
如果没指定引擎，表默认引擎是什么？
如果我不需要分析操作，是不是不会同步到AP中？

**Comment by leizzboy** [comment]
Quesion translate：
1) Suppose I create a table TestA in TP and indicate engine=innodb. Will this table be synchronized to AP after some transaction write operations? (would a table in TP also be in AP?)
2) If I want to analyze TestA, do I need to wait for TP synchronization before I can analyze it?
3) Will all tables in TP be synchronized to AP?
4) If no engine is specified, what is the default engine of the table?
5) If I don't need analysis operations, don't I need to synchronize to the AP?

**Comment by leizzboy** [comment, marked as answer]
Answer：
1) If the storage engine engine=innodb is specified when creating a table on the TP side, the table will be synchronized to the AP side, but its storage engine on the AP side would be InnoDB.
2) If you continue to analyze this table, you don't need to wait for the AP side to synchronize, and the analyze on the TP side will also be synchronized to the AP side. However, because this table is still an InnoDB engine on the AP side, it does not have the advantages of the stonedb storage engine.
3) Yes.
4) If no storage engine is specified for creating a table, the storage engine of the table depends on the parameter of default_storage_engine. It is recommended that the parameter of TP side be set to default_ storage_Engine=innodb, the parameter on the AP side is set to default_ storage_engine=stonedb。
5) The system will synchronize automatically. The TP side does analyze on the table, but the AP side does not support analyze on the table, and there is only a warning prompt.

回答：
1）如果在TP端创建表时指定了存储引擎engine=innodb，那么这张表会同步到AP端，但在AP端它的存储引擎还是innodb。
2）如果继续对这张表做analyze操作，不需要等AP端同步完，在TP端的analyze也会同步到AP端。但因为这张表在AP端还是innodb引擎，所以就没有stonedb存储引擎的特性。
3）是的。
4）如果创建表没有指定存储引擎，表的存储引擎取决于参数default_storage_engine的值。建议TP端的参数设置为default_storage_engine=innodb，AP端的参数设置为default_storage_engine=stonedb。
5）系统会自动同步。TP端对表做analyze操作，AP端是不支持对表进行analyze操作的，只会有warning提示。
